modules/rostests/apitests/user32/LoadImageGCC.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Test for LoadImageW using DLL compiled with MSVC
+ * COPYRIGHT:   Copyright 2024 Doug Lyons <douglyons@douglyons.com>
+ *
+ * NOTES:
+ * Works on ReactOS, but not on Windows 2003 Server SP2.
+ */
+
+#include "precomp.h"
+#include "resource.h"
+#include <stdio.h>
+#include <versionhelpers.h>
+
+WCHAR szWindowClass[] = L"testclass";
+
+BOOL FileExistsW(PCWSTR FileName)
+{
+    DWORD Attribute = GetFileAttributesW(FileName);
+
+    return (Attribute != INVALID_FILE_ATTRIBUTES &&
+            !(Attribute & FILE_ATTRIBUTE_DIRECTORY));
+}
+
+BOOL ResourceToFileW(INT i, PCWSTR FileName)
+{
+    FILE *fout;
+    HGLOBAL hData;
+    HRSRC hRes;
+    PVOID pResLock;
+    UINT iSize;
+
+    if (FileExistsW(FileName))
+    {
+        /* We should only be using %temp% paths, so deleting here should be OK */
+        printf("Deleting '%S' that already exists.\n", FileName);
+        DeleteFileW(FileName);
+    }
+
+    hRes = FindResourceW(NULL, MAKEINTRESOURCEW(i), MAKEINTRESOURCEW(RT_RCDATA));
+    if (hRes == NULL)
+    {
+        skip("Could not locate resource (%d). Exiting now\n", i);
+        return FALSE;
+    }
+
+    iSize = SizeofResource(NULL, hRes);
+
+    hData = LoadResource(NULL, hRes);
+    if (hData == NULL)
+    {
+        skip("Could not load resource (%d). Exiting now\n", i);
+        return FALSE;
+    }
+
+    // Lock the resource into global memory.
+    pResLock = LockResource(hData);
+    if (pResLock == NULL)
+    {
+        skip("Could not lock resource (%d). Exiting now\n", i);
+        return FALSE;
+    }
+
+    fout = _wfopen(FileName, L"wb");
+    fwrite(pResLock, iSize, 1, fout);
+    fclose(fout);
+    return TRUE;
+}
+
+static struct
+{
+    PCWSTR FileName;
+    INT ResourceId;
+} DataFiles[] =
+{
+    {L"%SystemRoot%\\bin\\image.dll", IDR_DLL_NORMAL},
+};
+
+/* MessageBox statements are left commented out for debugging purposes */
+static LRESULT CALLBACK
+WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    static HBITMAP hBmp;
+    HANDLE handle;
+
+    switch (message)
+    {
+        case WM_CREATE:
+        {
+            handle = LoadLibraryExW(L"image.dll", NULL, LOAD_LIBRARY_AS_DATAFILE);
```

Comment:
It is better to specify a full path here as well.

Reply:
Changed to use PathBuffer from above which is the full path. Thanks.